api/v1beta1/common_types.go

```diff
@@ -142,6 +142,10 @@ const (
 	// will receive the resulting object.
 	TopologyDryRunAnnotation = "topology.cluster.x-k8s.io/dry-run"
 
+	// SkipDefaultingWebhook is an annotation that indicates we don't want to mutate objects during admission
+	// Ref: https://cluster-api.sigs.k8s.io/developer/providers/webhooks.html#defaulting-webhooks
```

Comment:
@jackfrancis might be I'm missing something, but what is the use case for which we want to skip defaulting?

I'm asking because disabling a web-hook can lead to many unexpected edge-cases, and when we did something similar for the validation webhook we make sure this option works only when doing SSA dry-run calls (See [util func]( https://github.com/kubernetes-sigs/cluster-api/blob/bb287a1aabe1c1e54855441a10498086a9dd9683/util/topology/topology.go#L31)).

Reply:
Good question.

This is a draft PR that has some code to demonstrate how we might deal with this: https://github.com/kubernetes-sigs/cluster-api/issues/8668

tl;dr some gitops users don't want the yaml to be mutated upon admission, so that their yaml is able to be re-used as a source of truth

Reply:
🤔 Sharing the same concerns Fabrizio pointed out above, it seems very error prone to disable any sort of webhook logic, most importantly ones that are effectively treated as defaults in the reconciliation code